qutip/solver/multitraj.py

```diff
@@ -0,0 +1,464 @@
+from .. import Qobj, QobjEvo
+from .options import SolverOptions, SolverOdeOptions
+from .result import Result, MultiTrajResult
+from .parallel import get_map
+from time import time
+from .solver_base import Solver
+import numpy as np
+from copy import copy
+
+
+class MultiTrajSolver:
+    """
+    Basic class for multi-trajectories evolutions.
+
+    As :class:`Solver` it can ``run`` or ``step`` evolution.
+    It manage the random seed for each trajectory.
+
+    The actual evolution is done by a single trajectory solver::
+        ``_traj_solver_class``
+
+    Parameters
+    ----------
+    rhs : Qobj, QobjEvo
+        Right hand side of the evolution::
+            d state / dt = rhs @ state
+
+    options : SolverOptions
+        Options for the solver
+    """
+    _traj_solver_class = None
+    name = "generic multi trajectory"
+    optionsclass = SolverOptions
+    odeoptionsclass = SolverOdeOptions
+
+    def __init__(self, rhs, c_ops=(), *, options=None):
```

Comment:
I kind of want to do this everywhere.
`None` need to be converted to an empty list since we need to loop over them etc., but we can't default to a mutable `list`.
But an empty tuple does not need can be used to do the same and is safe.
I don't think that any user uses explicitly `c_ops=None`.